src/localization_plugin.cpp

```diff
@@ -242,6 +242,24 @@ void LocalizationPlugin::initialize()
 {
 }
 
+/**
+ * @brief fromPoseMsg create a 2dpose from the information in a posewithcovariance message
+ * @param msg
+ * @return
+ */
+geo::Transform2 fromPoseMsg(const geometry_msgs::PoseWithCovarianceStampedConstPtr& msg)
+{
+    tf2::Quaternion q;
+    tf2::convert(msg->pose.pose.orientation, q);
+
+    double yaw;
+    if (q.z() >= 0)
+        yaw = q.getAngle();
+    else
+        yaw = -q.getAngle();
+
+    return geo::Transform2(msg->pose.pose.position.x, msg->pose.pose.position.y, yaw);
+}
```

Comment:
Arguably this function should be moved to geolib

Reply:
Yes please. I am not completely familiar with quaternions, so where our calculations just wrong or is there a bug in our/someone else code.

P.S. We already have convert a function from `Pose` message to a Transform. And I think we project a Transform to a Transform2. But please check if the math is correct.

P.S. Next time you add a separate function, please place it either before or after all class functions.